File: alica-supplementary/plan-conversion-tool/src/main/java/com/rapyutarobotics/alica/factories/VariableFactory.java
package com.rapyutarobotics.alica.factories;

import com.rapyutarobotics.alica.Tags;
import de.unikassel.vs.alica.planDesigner.alicamodel.Variable;
import de.unikassel.vs.alica.planDesigner.modelmanagement.Types;
import org.w3c.dom.Element;

public class VariableFactory extends Factory {

    public static Variable create(Element variableNode) {
        Variable variable = new Variable();
        Factory.setAttributes(variableNode, variable);
        conversionTool.planElements.put(variable.getId(), variable);
        variable.setVariableType(variableNode.getAttribute(Tags.VARIABLETYPE));
        return variable;
    }
}

File: alica_engine/src/engine/modelmanagement/factories/VariableFactory.cpp
#include "engine/modelmanagement/factories/VariableFactory.h"
#include "engine/model/Variable.h"
#include "engine/modelmanagement/Strings.h"

namespace alica
{
Variable* VariableFactory::create(const YAML::Node& variableNode)
{
    Variable* variable = new Variable();
    Factory::setAttributes(variableNode, variable);
    Factory::storeElement(variable, alica::Strings::variable);

    variable->_type = Factory::getValue<std::string>(variableNode, alica::Strings::variableType, alica::Strings::no_type);

    return variable;
}

Variable* VariableFactory::create(int64_t id, const std::string& name, const std::string& type)
{
    Variable* variable = new Variable(id, name, type);
    Factory::storeElement(variable, alica::Strings::variable);

    return variable;
}
} // namespace alica
